app/controllers/workflow_controller.rb

```diff
@@ -23,11 +23,23 @@ def show_searchbar?
   end
 
   def button
-    if params[:pressed] == "embedded_configuration_script_payload_tag"
+    case params[:pressed]
+    when 'embedded_configuration_script_payload_map_credentials'
+      javascript_redirect(:action => 'map_credentials', :id => params[:miq_grid_checks])
+    when 'embedded_configuration_script_payload_tag'
       tag(self.class.model)
     end
   end
 
+  def map_credentials
+    assert_privileges('embedded_configuration_script_payload_map_credentials')
+    auth = self.class.model.find(params[:id])
```

Comment:
Is this the `id` of an authentication?  I would have thought it was the configuration_script_payload ID

Reply:
(a `Authentication.find(params[:id])` here scares me and is an issue, want to make sure this isn't that)

Reply:
I'm guessing we should be using `find_record_with_rbac` instead of `self.class.model.find()` anyway

Reply:
`self.class.model` = `ManageIQ::Providers::Workflows::AutomationManager::Workflow`

Reply:
Yeah okay so I think `auth` is a misleading variable name since this will be a `workflow` not an `authentication` (plus we should be using RBAC for the query)

Reply:
sorry I copied some of the code from other controllers so that might be a leftover

Reply:
Okay I'm not very familiar with the UI controller code so maybe it is okay and params are checked elsewhere?
@Fryguy ?

Reply:
I'm not sure I understand what the issue is here - I think I need to see it?

Reply:
@Fryguy is it an issue to do a `find(params[:id])` without RBAC in the UI?

Reply:
Oh the `auth` name was also throwing me off.  Yeah, I think this should use `find_record_with_rbac`

Reply:
Yeah I was stream of consciousness commenting ha sorry for the confusion, two issues I here
1. auth is a misleading variable name since this isn't an authentication
2. we should be using rbac not a raw `.find` on user provided parameters